editor/editor_node.cpp

```diff
@@ -872,7 +872,11 @@ void EditorNode::_resources_changed(const Vector<String> &p_resources) {
 		}
 
 		if (!res->editor_can_reload_from_file()) {
-			continue;
+			Ref<Script> scr = res;
+			// Scripts are reloaded via the script editor.
+			if (scr.is_null() || (scr.is_valid() && ScriptEditor::get_singleton()->get_open_scripts().has(scr))) {
```

Comment:
```suggestion
			if (scr.is_null() || ScriptEditor::get_singleton()->get_open_scripts().has(scr)) {
```
`is_null` and `is_valid` are mutually exclusive